src/scalar_impl.h

```diff
@@ -294,13 +305,130 @@ static void secp256k1_scalar_split_lambda(secp256k1_scalar *r1, secp256k1_scalar
  * Cryptography on Sensor Networks Using the MSP430X Microcontroller" (Gouvea, Oliveira, Lopez),
  * Section 4.3 (here we use a somewhat higher-precision estimate):
  * d = a1*b2 - b1*a2
- * g1 = round((2^272)*b2/d)
- * g2 = round((2^272)*b1/d)
+ * g1 = round((2^384)*b2/d)
+ * g2 = round((2^384)*(-b1)/d)
+ *
+ * (Note that 'd' is also equal to the curve order, n, here because [a1,b1] and [a2,b2]
+ * can be found as outputs of the Extended Euclidean Algorithm on inputs 'n' and 'lambda').
+ *
+ * The function below splits a in r1 and r2, such that
+ * - r1 + lambda * r2 == a (mod n)
+ * - either 0 <= r1 <= (a1 + a2 + 1) / 2 or n - (a1 + a2 + 1)/2 <= r1 < n.
+ * - either 0 <= r2 <= (-b1 + b2) / 2 or n - (-b1 + b2)/2 <= r2 < n.
```

Comment:
Maybe add here that this implies either `r1` or `-r1` is in range `[0..2^128-1]` (and similarly for `r2`), as that's what we ultimately care about.